app/dao/reports_dao.py

```diff
@@ -16,3 +19,21 @@ def create_report(report: Report) -> Report:
     db.session.commit()
 
     return report
+
+
+def get_reports_for_service(service_id: str, limit_days: int) -> List[Report]:
+    """
+    Get all reports for a service generated within the specified number of days.
+    Args:
+        service_id: The UUID of the service
+        limit_days: Number of days to look back (default: 30)
+    Returns:
+        List of Report objects sorted by requested_at (newest first)
+    """
+    query = Report.query.filter_by(service_id=service_id)
+
+    if limit_days is not None:
+        date_threshold = datetime.datetime.utcnow() - datetime.timedelta(days=limit_days)
```

Comment:
Pylance is telling me:

> The method "utcnow" in class "datetime" is deprecated
> Use timezone-aware objects to represent datetimes in UTC; e.g. by calling .now(datetime.timezone.utc)

Same thing in the tests

Reply:
It looks like it is going to be hard to switch to timezone-aware datetime objects because it's not backwards compatible with the timezone-naive datetimes we use everywhere else in the app. If we try to compare a tz-aware to a tz-naive datetime, that will throw a `TypeError` and could break things. Here's some sample code that shows the issue:
![Screenshot 2025-03-25 at 11 37 18 AM](https://github.com/user-attachments/assets/21f2e601-8bcc-4e78-990d-d69269cbddc8)

Reply:
Wow it got deprecated and they recommend using something that doesn't even return the same type? Why, python, why?